src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java

```diff
@@ -122,6 +122,45 @@ private DatasetServiceDAO.StudyInsert createStudyInsert(DatasetRegistrationSchem
     );
   }
 
+  /**
+   * This method takes an instance of a dataset registration schema and updates the dataset.
+   *
+   * @param user    The User creating these datasets
+   * @param files   Map of files, where the key is the name of the field
+   * @return List of created Datasets from the provided registration schema
+   */
+  public Dataset updateDataset(
+      Integer datasetId,
+      User user,
+      DatasetUpdate update,
+      Map<String, FormDataBodyPart> files) throws IOException, SQLException {
+
+    if (Objects.isNull(update.getName())) {
+      throw new BadRequestException("Dataset name is required");
+    }
+
+    if (Objects.isNull(update.getDacId())) {
+      throw new BadRequestException("DAC Id is required");
+    }
+
+    Map<String, BlobId> uploadedFileCache = new HashMap<>();
+
+    DatasetServiceDAO.DatasetUpdate datasetUpdates;
+
+    try {
+      datasetUpdates = createDatasetUpdate(datasetId, user, update, files, uploadedFileCache);
+
+    } catch (IOException e) {
+      // uploading files to GCS failed. rollback files...
+      uploadedFileCache.values().forEach((id) -> gcsService.deleteDocument(id.getName()));
+      throw e;
+    }
+
+    // Update or create the objects in the database
+    datasetServiceDAO.updateDataset(datasetUpdates);
+    return datasetDAO.findDatasetById(datasetId);
```

Comment:
This could all be inside the try so that you don't have to declare, then populate, `datasetUpdates`